src/sentry/reprocessing2.py

```diff
@@ -347,27 +347,33 @@ def build_event_key(primary_hash):
         event_count += client.llen(key)
 
     if force_flush_batch or event_count > settings.SENTRY_REPROCESSING_REMAINING_EVENTS_BUF_SIZE:
-        for primary_hash in old_primary_hashes:
-            event_key = build_event_key(primary_hash)
-            events = client.lrange(event_key, 0, -1)
-            event_ids, from_date, to_date = calculate_date_range_and_get_event_ids(events)
-
-            if len(event_ids) == 0:
-                continue
-
-            from sentry import eventstream
-
-            # In the worst case scenario, a group will have a 1:1 mapping of primary hashes to
-            # events, which means 1 insert per event.
-            # The overall performance of this will be marginally better than the unbatched version
-            # if a group has a lot of old primary hashes.
-            eventstream.tombstone_events_unsafe(
-                project_id,
-                event_ids,
-                old_primary_hash=old_primary_hash,
-                from_timestamp=from_date,
-                to_timestamp=to_date,
-            )
+        with sentry_sdk.start_span(
+            op="sentry.reprocessing2.buffered_delete_old_primary_hash.flush_events"
+        ):
+            for primary_hash in old_primary_hashes:
+                event_key = build_event_key(primary_hash)
+                event_ids, from_date, to_date = pop_batched_events_from_redis(event_key)
+
+                if len(event_ids) == 0:
+                    with sentry_sdk.push_scope() as scope:
+                        scope.set_tag("project_id", project_id)
+                        scope.set_tag("old_group_id", group_id)
+                        scope.set_tag("old_primary_hash", old_primary_hash)
+                        raise CannotReprocess("events_batch.not_found")
```

Comment:
I would do `logger.error("events_batch.not_found", project_id=project_id, ...)`. I think there should be a logger instantiated somewhere in this module. Raising an exception here will have it bubble up and wreck things that you may not want to see wrecked (since this runs in `save_event`)

Also, your tags will not be applied here, because when the exception is bubbling up, the scope is popped again. This would work:

```
with push_scope():
    set_tag(...)
    capture_exception(CannotReprocess("events_batch.not_found"))
```

or this

```
with push_scope():
    set_tag(...)
    try:
        raise ...
    except ...:
        capture_exception()
```